internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1705,27 +1656,41 @@ func BootstrapEtcd(seq runtime.Sequence, data interface{}) (runtime.TaskExecutio
 			}
 		}
 
-		// Since etcd has already attempted to start, we must delete the data. If
-		// we don't, then an initial cluster state of "new" will fail.
-		dir, err := os.Open(constants.EtcdDataPath)
-		if err != nil {
-			return err
-		}
+		if err = func() error {
+			// Since etcd has already attempted to start, we must delete the data. If
+			// we don't, then an initial cluster state of "new" will fail.
+			var dir *os.File
 
-		//nolint:errcheck
-		defer dir.Close()
+			dir, err = os.Open(constants.EtcdDataPath)
+			if err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
 
-		files, err := dir.Readdir(0)
-		if err != nil {
-			return err
-		}
+				return err
+			}
+
+			//nolint:errcheck
+			defer dir.Close()
 
-		for _, file := range files {
-			fullPath := filepath.Join(constants.EtcdDataPath, file.Name())
+			var files []os.FileInfo
 
-			if err = os.RemoveAll(fullPath); err != nil {
-				return fmt.Errorf("failed to remove %q: %w", file.Name(), err)
+			files, err = dir.Readdir(0)
```

Comment:
I wonder if we can remove the whole `constants.EtcdDataPath` instead. `if os.IsNotExist(err) { return nil }` above means that some other code creates this directory.

Reply:
I don't remember what was the reason to put it this way, but I remember there was some. Probably @andrewrynhard might know, but once again deserves a separate PR even if it is unneccessary.

Reply:
I don't quite remember why we had to do it this way but I do remember being frustrated with having to do it this way. But worth a try in another PR to clean it up.